src/luarocks/fs/unix.lua

```diff
@@ -145,13 +145,14 @@ function unix.is_actual_binary(filename)
    if not file then
       return true
    end
-   local first = file:read(2)
+   local first = file:read()
    file:close()
    if not first then
       util.warning("could not read "..filename)
       return true
    end
-   return first ~= "#!"
+   -- only create wrapper for lua scripts
+   return first:match('#!.* lua') == first
```

Comment:
This is not an effective robust match. The space before `lua` is optional (it might be part of a full path), the capitalization could differ, a space could be used after the hash and before the bang, this may not be the end of the line, etc.